lightning/src/ln/channel.rs

```diff
@@ -4720,6 +4767,18 @@ impl<Signer: Sign> Channel<Signer> {
 				}
 			}
 		}
+		if self.is_outbound() {
+			// Log if we can't afford next remote commitment tx fee at pending outbound feerate update.
+			if let Some(pending_feerate) = self.pending_update_fee {
+				assert_eq!(pending_feerate.1, FeeUpdateState::Outbound);
+				let next_total_fee_sat = Channel::<Signer>::commit_tx_fee_sat(pending_feerate.0, self.pending_inbound_htlcs.len() + self.pending_outbound_htlcs.len());
+				let outbound_stats = self.get_outbound_pending_htlc_stats(Some(pending_feerate.0));
+				let holder_balance_after_fee_sub_sats = (self.value_to_self_msat / 1000).checked_sub(next_total_fee_sat).map(|a| a.checked_sub(outbound_stats.pending_htlcs_value_msat / 1000)).unwrap_or(None).unwrap_or(u64::min_value());
+				if holder_balance_after_fee_sub_sats < self.counterparty_selected_channel_reserve_satoshis.unwrap() {
+					log_debug!(logger, "Outbound update_fee HTLC buffer overflow - counterparty should force-close this channel");
```

Comment:
Why can't we just generate a second `UpdateFee` here which overwrites the previous one and re-reduces the fee back to where we can afford it?

Reply:
I checked the spec on that, it doesn't forbid to send multiple `update_fee` in the same sequence, however it doesn't make any prescription on multiple `update_fee` so I'm not sure how well it processed by other implementations.

But beyond, I think we can always fall-back in some kind of loop, where we cancel `update_fee` to an affordable rate, a concurrent HTLC is received, the now-reduced affordable rate isn't relevant anymore, so we need to cancel even lower, etc ?

If you're really thinking about it, I would like to think more we don't introduce weird halting or freezing in the state machine, potentially triggerable by the counterparty and do that in a separate issue.

Reply:
Hmm, I'd think nodes handle it, I guess we'd want to test, but it seems strictly preferable to the channel being closed? At least if we drop the feerate back down to where it was before we updated it either the counterparty closes due to bogus feerate or they close due to incorrect signature and we're no worse off than just not updating the fee to begin with.

Reply:
That said, I'm not sure we need to care about this - we generally update the fee and commit under the same lock, so I think in fact this code may be entirely unreachable. Why would we ever hit this after returning success in `update_fee`?

Reply:
After thinking more, I agree with you. And same if we try to `send_htlc`, we'll hit fee-affordability checks there. And if we our counterparty send us a concurrent inbound HTLC we won't take this path.

Removed.